dot/sync/interface.go

```diff
@@ -54,9 +54,10 @@ type BlockState interface {
 // StorageState is the interface for the storage state
 type StorageState interface {
 	TrieState(root *common.Hash) (*rtstorage.TrieState, error)
-	StoreTrie(ts *rtstorage.TrieState, header *types.Header) error
 	LoadCodeHash(*common.Hash) (common.Hash, error)
 	SetSyncing(bool)
+	BeginModifyTrie()
+	FinishModifyTrie()
```

Comment:
```suggestion
	Lock()
	Unlock()
```
you embedded the `sync.Mutex` so it should fulfil the interface

Reply:
nice, updated!